### PR TITLE
Added new tests nad fixes

### PR DESCRIPTION
**Pull request by szymon-rd**

### CODE REVIEW

**Comment by szymon-rd** [comment]
Fixing failing tests on windows...

**Comment by szymon-rd** [comment]
Extracted issue of two failing tests on WIndows to #40

File: src/test/java/com/virtuslab/using_directives/parser/ParserNumericTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  public void testHexNegativeNumeric() {
+    String input = "using keyA -0xA2";
+    UsingDirectives parsedDirective = testCode(1, input);
+    assertValueAtPath(parsedDirective, "keyA", "-0xA2");
+  }
+
+  @Test
+  public void testFailTrailingSeparator() {
+    PersistentReporter reporter = reporterAfterParsing("using key 200_");
+    assertDiagnostic(reporter, 0, 13, "Trailing separator is not allowed");
```
Comment (romanowski) [inline src/test/java/com/virtuslab/using_directives/parser/ParserNumericTest.java:86]:
```suggestion
    assertDiagnostic(reporter, 0, 13, "Trailing number separator (`_`) is not allowed");
```

File: src/test/java/com/virtuslab/using_directives/parser/ParserScopeTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    String input = "using keyA \"fooA\", \"fooB\" in \"scopeA\"";
+    UsingDirectives parsedDirective = testCode(1, input);
+    assertValueListSizeAtScope(parsedDirective, "keyA", "scopeA", 2);
+  }
+
+  @Test
+  public void testFailScopeNotAString() {
+    String input = "using keyA \"fooA\", \"fooB\" in 2";
+    PersistentReporter reporter = reporterAfterParsing(input);
+    assertDiagnostic(reporter, 0, 29, "Expected token STRINGLIT but found", "integer literal");
```
Comment (romanowski) [inline src/test/java/com/virtuslab/using_directives/parser/ParserScopeTest.java:32]: IMHO this message is really, critpic. I think it should say `Scope should be a string literal, did you forget to quote?`

File: src/test/java/com/virtuslab/using_directives/parser/ParserScopeTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  @Test
+  public void testFailScopeMissing() {
+    String input = joinLines("using", "   keyA \"fooA\" in ", "   keyA \"fooB\" in \"scopeB\"");
+    PersistentReporter reporter = reporterAfterParsing(input);
+    assertDiagnostic(reporter, 2, 3, "Expected token STRINGLIT but found");
```
Comment (romanowski) [inline src/test/java/com/virtuslab/using_directives/parser/ParserScopeTest.java:39]: This one should say something about empty scope

File: src/test/java/com/virtuslab/using_directives/parser/ParserUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  public void testBooleanLiteral() {
+    String input = "using keyA false";
+    UsingDirectives parsedDirective = testCode(1, input);
+    assertValueAtPath(parsedDirective, "keyA", "false");
+  }
+
+  @Test
+  public void testFailMissingBrace() {
+    PersistentReporter reporter = reporterAfterParsing("using keyA {", "   keyB 18", "x");
+    assertDiagnostic(reporter, 2, 0, "Expected closing region token but found identifier");
```
Comment (romanowski) [inline src/test/java/com/virtuslab/using_directives/parser/ParserUnitTest.java:36]: `closing regin` is a bit criptic, can we tell at the point if an indent or brace is missing?

File: src/test/java/com/virtuslab/using_directives/parser/ParserUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  @Test
+  public void testFailExpectingIndentOrBrace() {
+    PersistentReporter reporter = reporterAfterParsing("using (keyB 2)");
+    assertDiagnostic(reporter, 0, 6, "Expected indent or braces but found", "(");
```
Comment (romanowski) [inline src/test/java/com/virtuslab/using_directives/parser/ParserUnitTest.java:42]: I do not thing that we use parentheses anywhere in our syntax so I think we can say that are not supported.

File: README.md [hunk 6]
```diff
@@ -1,3 +1,10 @@
...
-Library for processing meta-information written in using-directives syntax.
\ No newline at end of file
+Library for processing meta-information written in using-directives syntax.
+
+### Generating code coverage report
+Run:
+```shell
+./gradlew jacocoTestReport
+```
+Then, the coverage report will appear in the output directory, inside the `coverage_report` directory.
```
Comment (romanowski) [inline README.md:10]: I think we should include guidelines what is expected coverage so developers can compare their results

**Review by romanowski — APPROVED** [review]
Most of the comments apply to error messages that are outside of the scope of this PR so I think we can improve it as a follow-up.

**Pull request merged**